AADUserInfo Function

dicolanl opened the pull request:
Fixes #
#961  

## Proposed Changes
Function app to pull in AAD User Info daily.

v-ampami commented on DataConnectors/AADUserInfo/azuredeploy.json:
```diff
@@ -0,0 +1,251 @@
+                        "WEBSITE_CONTENTSHARE": "[toLower(parameters('FunctionName'))]",
+                        "workspaceID": "[parameters('workspaceID')]",
+                        "workspaceKey": "[concat('@Microsoft.KeyVault(SecretUri=', reference(resourceId('Microsoft.KeyVault/vaults/secrets', parameters('FunctionName'), 'workspaceKey')).SecretUriWithVersion, ')')]",
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/AADUserInfo/AADUserInfo_template.zip?raw=true"                    }
```
Please replace WEBSITE_RUN_FROM_PACKAGE value with [https://aka.ms/sentinel-aaduserinfo-functionapp](https://aka.ms/sentinel-aaduserinfo-functionapp)

v-ampami commented on Parsers/AADUserInfo/AADUserInfo.txt:
```diff
@@ -0,0 +1,40 @@
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias as AADUserInfo.
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. AADUserInfo | take 10).
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+AADUserInfo_CL
```
Sample data not provided. Getting error ("'summarize' operator: Failed to resolve table or column expression named 'AADUserInfo_CL'") in parser deployment in log analytics workspace. 

Please fix the issue.

dicolanl replied:
added CL schema

v-ampami commented on DataConnectors/AADUserInfo/AADUserInfo/run.ps1:
```diff
@@ -0,0 +1,83 @@
+}
+
+# Get Managed Service Identity info from Azure Functions Application Settings
+$msiEndpoint = $env:MSI_ENDPOINT
```
$msiEndpoint, $msiSecret variables are not used anywhere in the code or taken from input. Can you please remove these variable if you are not using them.

v-ampami commented:
@dicolanl Please update the following requested changes:

Please move these files to [Solution folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions). Follow this [folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box) for reference.

Please include DataConnector file in PR. Refer to this [https://github.com/Azure/Azure-Sentinel/tree/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template](https://github.com/Azure/Azure-Sentinel/tree/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template) for reference.

Please update changes related to Azure gov. Refer to this [PRLink](https://github.com/Azure/Azure-Sentinel/pull/1762/files) for gov changes.

dicolanl commented:
> @dicolanl Please update the following requested changes:
> 
> Please move these files to [Solution folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions). Follow this [folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box) for reference.
> 
> Please include DataConnector file in PR. Refer to this https://github.com/Azure/Azure-Sentinel/tree/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template for reference.
> 
> Please update changes related to Azure gov. Refer to this [PRLink](https://github.com/Azure/Azure-Sentinel/pull/1762/files) for gov changes.

moved to solutions folder
added sample data
added _CL schema

v-ampami commented:
> > @dicolanl Please update the following requested changes:
> > Please move these files to [Solution folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions). Follow this [folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box) for reference.
> > Please include DataConnector file in PR. Refer to this https://github.com/Azure/Azure-Sentinel/tree/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template for reference.
> > Please update changes related to Azure gov. Refer to this [PRLink](https://github.com/Azure/Azure-Sentinel/pull/1762/files) for gov changes.
> 
> moved to solutions folder
> added sample data
> added _CL schema

Please include DataConnector file in PR. Refer to this https://github.com/Azure/Azure-Sentinel/tree/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template for reference.

Please fix the PR Validation issues.

v-ampami commented on Solutions/AADUserInfo/Data Connectors/AADUserInfo/run.ps1:
```diff
@@ -0,0 +1,96 @@
+        -resource $resource
+    
+        # Compatible with previous version
+		if ([string]::IsNullOrEmpty($logAnalyticsUri)){
```
Please update the logAnalyticsUri code  referring to this [PRLink](https://github.com/Azure/Azure-Sentinel/pull/1762/files).

dicolanl replied:
@v-ampami. this code looks right.  Line 76 has the env variable, and lines 78-84 check that its a valid url.  this is like many of the other functions already approved?

v-ampami commented on Solutions/AADUserInfo/Data Connectors/README.MD:
```diff
@@ -0,0 +1,83 @@
+11. Click Save.
+
+If successfully deployed you should start to see events appear in your Azure Sentinel workspace as soon as they are generated.
+If you run into issues there are a number of options for [monitoring](https://docs.microsoft.com/en-us/azure/azure-functions/functions-monitoring?tabs=cmd) and [deugging](https://docs.microsoft.com/en-us/azure/azure-functions/functions-debug-powershell-local) your Function App.
```
This documentation link has en-us locale that is causing the Validation to fail. Please update the link.

v-ampami commented on .vscode/tasks.json:
```diff
@@ -0,0 +1,14 @@
+{
```
Please remove this file from PR.

v-ampami commented on Solutions/AADUserInfo/Data Connectors/AADUserInfo/run.ps1:
```diff
@@ -0,0 +1,96 @@
+    Write-Host "PowerShell timer is running late!"
+}
+
+Import-Module AzureAD -UseWindowsPowerShell
```
Getting error while testing. Please fix the issue.
![image](https://user-images.githubusercontent.com/78137909/121034050-577c0980-c7ca-11eb-9827-5a66719bfb46.png)

dicolanl replied:
How are you testing this? in Azure Functions?  I had to add -UseWindowsPowershell to get it to work in Azure Functions.  There is even a GH issue (here)[https://github.com/Azure/azure-functions-powershell-worker/issues/232] that says to do this in azure functions
